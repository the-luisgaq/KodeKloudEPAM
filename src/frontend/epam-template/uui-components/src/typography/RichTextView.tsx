import * as React from 'react';
import '../index.scss';

// console.log(css)
import {
    IHasCX, IHasChildren, IHasRawProps, cx,
} from '@epam/uui-core';

export interface RichTextViewProps extends IHasCX, IHasChildren, IHasRawProps<React.HTMLAttributes<HTMLDivElement>> {
    /** HTML content to be rendered */
    htmlContent?: any;
}

export class RichTextView extends React.Component<RichTextViewProps> {
    getViewByProps(content: string) {
        return <div dangerouslySetInnerHTML={ { __html: content } } className={ cx(this.props.cx) } { ...this.props.rawProps } />;
    }

    getViewByChildren(content: React.ReactNode) {
        return (
            <div className={ cx(this.props.cx) } { ...this.props.rawProps }>
                {content}
            </div>
        );
    }

    render() {
        let content = this.getViewByChildren(this.props.children);

        if (this.props.htmlContent) {
            content = this.getViewByProps(this.props.htmlContent);
        }

        return content;
    }
}
