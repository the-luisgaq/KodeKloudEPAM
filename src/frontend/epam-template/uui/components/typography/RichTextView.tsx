import * as uuiComponents from '@epam/uui-components';
import type { Overwrite } from '@epam/uui-core';
import { withMods } from '@epam/uui-core';
import { settings } from '../../settings';

interface RichTextViewMods {
    /**
     * Defines component size.
     * @default '14'
     */
    size?: '12' | '14' | '16';
}

export interface RichTextViewModsOverride {}

export interface RichTextViewProps extends uuiComponents.RichTextViewProps, Overwrite<RichTextViewMods, RichTextViewModsOverride> {}

export const RichTextView = withMods<uuiComponents.RichTextViewProps, RichTextViewProps>(
    uuiComponents.RichTextView,
    (mods: RichTextViewMods) => ['uui-typography', `uui-typography-size-${mods.size || settings.richTextView.sizes.default}`],
);
