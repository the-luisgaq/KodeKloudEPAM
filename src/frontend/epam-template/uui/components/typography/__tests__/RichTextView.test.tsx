import React from 'react';
import { render } from '@epam/uui-test-utils';
import { RichTextView } from '../RichTextView';

describe('RichTextView', () => {
    it('should be rendered correctly', () => {
        const { asFragment } = render(
            <RichTextView>
                <p>Test</p>
            </RichTextView>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should be rendered correctly', () => {
        const { asFragment } = render(
            <RichTextView size="16">
                <div>Test</div>
            </RichTextView>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
