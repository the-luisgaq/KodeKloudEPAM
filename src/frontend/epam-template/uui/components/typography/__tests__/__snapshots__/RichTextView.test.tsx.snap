// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichTextView should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="uui-typography uui-typography-size-14"
  >
    <p>
      Test
    </p>
  </div>
</DocumentFragment>
`;

exports[`RichTextView should be rendered correctly 2`] = `
<DocumentFragment>
  <div
    class="uui-typography uui-typography-size-16"
  >
    <div>
      Test
    </div>
  </div>
</DocumentFragment>
`;
